User admin page loading.

emcelroy opened the pull request:
PT Story: https://www.pivotaltracker.com/story/show/176874032

[#176874032]

Limit the number of users returned to 30 since we only show thirty per page.

scytacki reviewed (changes requested):
Have you checked the SQL if this makes a difference?
It seems like the searchable_model which I think is what is used when `search` is called, has page and per_page filters. I think these should basically be doing the same limit and offset that you are doing in this PR:
https://github.com/concord-consortium/rigse/blob/668932310dc2c9f7c1e6cf39cf50e883807a32a7/rails/lib/searchable_model.rb#L69

emcelroy commented:
@scytacki I may be missing something in your question and/or the code, sorry, but this is what I've tested:

With the limit and offset I added, `search_scope` is a collection of 30 users when the users index page first loads. If I remove the limit and offset, `search_scope` is a collection of every user in my local portal.

With the limit and offset I added, the SQL command for `search_scope` looks like this when the page first loads (without any params):

`SELECT  DISTINCT `users`.* FROM `users` LEFT JOIN roles_users ON users.id = roles_users.user_id LEFT JOIN admin_project_users ON users.id = admin_project_users.user_id WHERE (`users`.`state` != 'disabled') LIMIT 30 OFFSET 0`

Without the limit and offset, it looks like this:

`SELECT DISTINCT `users`.* FROM `users` LEFT JOIN roles_users ON users.id = roles_users.user_id LEFT JOIN admin_project_users ON users.id = admin_project_users.user_id WHERE (`users`.`state` != 'disabled')`

Does that answer your question? I thought limiting the number of users in the scope when the users page first loads made sense, but is that not a good idea?
